### PR TITLE
[Doc] Adding spamtrap module

### PR DESCRIPTION
**Pull request by croessner**

### CODE REVIEW

File: doc/modules/spamtrap.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+### Module settings
+
+Parameters for the spamtrap modules are listed here.
+
+- `action`: You can optionally set an action
+- `symbol`: The name of a symbol that will be inserted, if a match between
+  recipient and a spam trapped email/domain was found. Defaults to 'SPAMTRAP'
+- `score`: The score for this symbol. It defaults to neutral 0.0
+- `learn_fuzzy`: Boolean that enables or disables fuzzy learning. Defaults to
```
Comment (fatalbanana) [inline doc/modules/spamtrap.md]: This (and `learn_spam`) are actually default true in the plugin.

Comment (croessner) [inline doc/modules/spamtrap.md]: Yes. But the default module config defaults to false 😉

Comment (fatalbanana) [inline doc/modules/spamtrap.md]: Aha. I see.

File: doc/modules/spamtrap.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+With the spamtrap module you can catch spam trapped emails or even a catch-all domain.
+It is intended for learning bayes spam and, if you run your own fuzzy storage, for learning
+a fuzzy flag with a fuzzy weight. You can also add a score for this symbol in the metrics,
+if you only want to tag such emails. You can use this later in other modules like
+force_actions or metadata_exporter. Examples will be given below.
+
+## Spamtrap setup
+
+Spamtrap currently requires redis being configured and used. The idea is to add all your
```
Comment (fatalbanana) [inline doc/modules/spamtrap.md]: Seems to be stale.

File: doc/modules/spamtrap.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+~~~
+
+An example of a map file is:
+
+~~~text
+/^test@example\.com$/
+/^.\*@catchalldomain.test$/
+~~~
+
+The first is a full email addressm while the second is a catch-all domain.
```
Comment (fatalbanana) [inline doc/modules/spamtrap.md]: Typo.

File: doc/modules/spamtrap.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+# Return spamtrap e-mail addresses from Redis
+settings_redis {
+        handlers = {
+                SPAMTRAP = <<EOD
+return function(task)
+        local rcpt = task:get_recipients('smtp')
+        if not (rcpt and #rcpt == 1) then
+                return
+        end
+        return 'sptr_' .. rcpt[1]['addr']:lower()
```
Comment (fatalbanana) [inline doc/modules/spamtrap.md]: Perhaps a SPAMTRAP_DOMAIN example would be helpful. It's not immediately clear what the point of this is & it seems to be somewhat conflated with the next point ("adding emails and domains to redis").

**Comment by croessner** [comment]
Hope, things are ok now. Updated and push --forced

**Pull request merged**